core/transfer/local/pull.go

```diff
@@ -296,10 +301,23 @@ func getSupportedPlatform(uc transfer.UnpackConfiguration, supportedPlatforms []
 			// Assume sp.SnapshotterKey is not empty
 			if uc.Snapshotter == sp.SnapshotterKey {
 				return true, sp
-			} else if uc.Snapshotter == "" && sp.SnapshotterKey == defaults.DefaultSnapshotter {
-				return true, sp
+			} else if uc.Snapshotter == "" {
+				if sp.SnapshotterKey == defaults.DefaultSnapshotter {
+					// Return as best match immediately
+					return true, sp
+				} else if !matched {
+					// Match but prefer default if present to prevent configuration
+					// changes from altering default behavior
+					matched = true
+					u = sp
+				}
 			}
+		} else if uc.Snapshotter == sp.SnapshotterKey {
+			log.G(ctx).WithFields(log.Fields{
+				"platform":    platforms.FormatAll(uc.Platform),
+				"snapshotter": uc.Snapshotter,
+			}).Info("Found unpack with matching snapshotter but platform does not match")
```

Comment:
Is this something the customer should be warned about (could cause an issue)?

Reply:
See #11606 as an example. There may be a follow-up after this change to correct for more cases or another bug which is causing that error.